File: Week 6 - Python/sentimental-readability/readability.py
# TODO

# Main function
def main():
    # Get text
    text = input("Text: ")
    # Declare variables
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    # Calculate Coleman-Liau index
    L = 100 * (letters / words)
    S = 100 * (sentences / words)
    index = round(0.0588 * L - 0.296 * S - 15.8)
    # Print grade
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Count number of letters in text
def count_letters(text):
    # Set counter
    countlet = 0
    for c in text:
        # If character in text is an alphabet
        if (c.isalpha()) == True:
            # Add 1 to counter
            countlet += 1
    return countlet


# Count number of words in text
def count_words(text):
    # Set counter
    countwor = 1
    for c in text:
        space = 32
        # If character in text is a space
        if c == chr(space):
            # Add 1 to counter
            countwor += 1
    return countwor


# Count number of sentences in text
def count_sentences(text):
    # Set counter
    countsen = 0
    for c in text:
        exclam = 33
        period = 46
        quest = 63
        # If character in text is an exclamation mark, question mark or period
        if c == chr(exclam) or c == chr(period) or c == chr(quest):
            # Add 1 to counter
            countsen += 1
    return countsen


main()

File: Week 2 - Arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get text
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculate Coleman-Liau index
    float L = 100 * ((float)letters / (float)words);
    float S = 100 * ((float)sentences / (float)words);
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Display grade
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Count number of letters
int count_letters(string text)
{
    int let = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        //Making sure only alphabets are counted
        if (isalpha(text[i]))
        {
            let++;
        }
    }
    return let;
}

// Count number of words
int count_words(string text)
{
    int wor = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Instead of counting words, count number of spaces + 1
        if (isspace(text[i]) > 0)
        {
            wor++;
        }
    }
    return wor;
}

// Count number of sentences
int count_sentences(string text)
{
    int sen = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Instead of counting sentence, count number of .'s, !'s or ?'s.
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sen++;
        }
    }
    return sen;
}


